tools/waveica/waveica_wrapper.R

```diff
@@ -10,25 +10,28 @@ waveica <- function(
 ) {
 
     # get input from the Galaxy, preprocess data
-    data <- read.csv(data, header = TRUE, row.names = "sampleName")
+    data <- read.csv(data, header = TRUE)
+
+    required_columns <- c("sampleName", "class", "sampleType", "injectionOrder", "batch")
+    if (anyNA(data)) {
+        stop("Error: dataframe cannot contain NULL values!
+Make sure that your dataframe does not contain empty cells")
+    }
+    else if (!all(required_columns %in% colnames(data))) {
+        stop("Error: missing metadata!
+Make sure that the following columns are present in your dataframe: [sampleName, class, sampleType, injectionOrder, batch]")
+    }
 
     # sort data by injection order
     data <- data[order(data$injectionOrder, decreasing = FALSE), ]
 
-    data <- enumerate_groups(data)
-
-    # remove blanks from dataset
-    if (exclude_blanks) {
-        data <- exclude_group(data)
-    }
-
     # separate data into features, batch and group
-    features <- data[, -c(1:4)]
-    group <- as.numeric(data$class)
+    features <- data[, -c(1:5)]
```

Comment:
Again, selecting by name might be more robust.

Reply:
Fixed in 2413166.